Lib/fontmake/instantiator.py

```diff
@@ -224,21 +226,33 @@ def from_designspace(
         # because the math behind varLib and MutatorMath uses the default font as the
         # point of reference for all data.
         default_font = designspace.default.font
+        non_default_layer_name = designspace.default.layerName
+        non_default_layer_flag = (
+            non_default_layer_name is not None
+            and non_default_layer_name != DEFAULT_LAYER_NAME
+        )
+
         glyph_names: Set[str] = set(default_font.keys())
 
-        for source in designspace.sources:
-            other_names = set(source.font.keys())
-            diff_names = other_names - glyph_names
-            if diff_names:
-                logger.warning(
-                    "The source %s (%s) contains glyphs that are missing from the "
-                    "default source, which will be ignored: %s. If this is unintended, "
-                    "check that these glyphs have the exact same name as the "
-                    "corresponding glyphs in the default source.",
-                    source.name,
-                    source.filename,
-                    ", ".join(sorted(diff_names)),
-                )
+        if non_default_layer_flag:
+            if non_default_layer_name in default_font.layers:
+                layer = default_font.layers[non_default_layer_name]
+                glyph_names = layer.keys()
+                logger.info(f"Building from layer {layer.name}")
+        else:
+            for source in designspace.sources:
+                other_names = set(source.font.keys())
+                diff_names = other_names - glyph_names
+                if diff_names:
+                    logger.warning(
```

Comment:
I think this check that there aren't different glyphs in non-default masters, we would like to do regardless of whether the default master is taken from a UFO's default layer or a non-default layer, so we should move the whole for block one indentation level up

Reply:
> This makes the build extremely noisy

I don't understand how this should be any different whether we use the default layer or a non-default one. If you think it's too noisy, we can print the first 10 or so and truncate with ellipsis.

Reply:
the effect of having additional glyphs in the non-default sources that are not in the default one is the same, whether or not the default source is taken from the default layer or a non-default layer: those glyphs will be ignored. I think it still warrants a warning.

Reply:
The above logic is what I just checked in for review.